Makefile

```diff
@@ -5,18 +5,37 @@
 
 sagefile := .sage/bin/sagefile
 
+# Setup Go.
+go := $(shell command -v go 2>/dev/null)
+ifndef go
+SAGE_GO_VERSION ?= 1.18.4
+export GOROOT := .sage/tools/go/$(SAGE_GO_VERSION)/go
+export PATH := $(PATH):$(GOROOT)/bin
+go := $(GOROOT)/bin/go
```

Comment:
Is this necessary? We are always adding bin to the PATH.

Reply:
Ah yes! This is just so that we can use it as a Make target further down - this is how we ensure that we detect if we have already downloaded Go (the `$(go)` target matches a file and the download is skipped).